openedx/features/journals/views/learner_dashboard.py

```diff
@@ -0,0 +1,101 @@
+""" Journal Tab of Learner Dashboard views """
+from datetime import datetime, time
+from django.conf import settings
+from django.http import Http404
+from urlparse import urljoin, urlsplit, urlunsplit
+
+from edxmako.shortcuts import render_to_response
+from openedx.core.djangoapps.programs.models import ProgramsApiConfig
+from openedx.features.journals.api import fetch_journal_access, journals_enabled
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def journal_listing(request):
+    """ View a list of journals which the user has or had access to"""
+
+    user = request.user
+
+    if not journals_enabled() or not user.is_authenticated():
+        raise Http404
+
+    journals = fetch_journal_access(
+        site=request.site,
+        user=request.user
+    )
+
+    context = {
+        'journals': journals,
```

Comment:
Not important for MVP, but if a learner renews their subscription, both will be returned as separate items here and be displayed as separate items on the dashboard. We might want to put in a future ticket to cluster accesses by Journal then display the current one if there is one, and if not, a renew link.